### PR TITLE
Don't overwrite auth header

### LINKED ISSUE #164: OAuth library access_token not getting passed in, causing '403 Not Authorized'
I am having trouble using the OAuth libraries making a second call to get client.users.me.  I traced the error down to the fact that no Authorization: Bearer headers were being passed in the GET request.

Digging deeper, it looks like the OAuthAuthenticator class is putting the headers in, but then they are getting blown over by these lines of code in the dispatcher.  This may be a recent change?  I am new to the library.

```
// from lib/dispatcher.js 0.16
    function doRequest() {
      if (me.authenticator !== null) {
        me.authenticator.authenticateRequest(params); // <-- OAuthAuthenticator adds Bearer to params.headers
      }
      params.headers = _.extend(
          {}, me.defaultHeaders, dispatchOptions.headers || {});   // <-- Blows over Bearer because params.headers is reassigned.
      me._addVersionInfo(params);

```

My temporary solution is to pass in my own dispatchOptions.   I manually calculate the headers and pass it in.  
```
// this does not work - I get a 403
client.users.me(); 

// but this works
const authHeaders = client.dispatcher.authenticator.authenticateRequest({});
client.users.me(null, authHeaders); 

```
 Am I missing something obvious...  It doesn't seem like I should have to pass authHeaders in?

### PR DESCRIPTION
**Pull request by joetrollo**
Fixes #164

### CODE REVIEW

File: lib/dispatcher.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ Dispatcher.prototype.dispatch = function(params, dispatchOptions) {
         me.authenticator.authenticateRequest(params);
```
Comment (praecipula) [inline lib/dispatcher.js]: Hmm, interesting. It took me a second to grok how the headers are treated on these lines, but I'm OK with the behavior. What happens here is that we set the auth header...

File: lib/dispatcher.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ Dispatcher.prototype.dispatch = function(params, dispatchOptions) {
...
       }
       params.headers = _.extend(
-          {}, me.defaultHeaders, dispatchOptions.headers || {});
+          {}, params.headers, me.defaultHeaders, dispatchOptions.headers || {});
```
Comment (praecipula) [inline lib/dispatcher.js]: Then we set the default headers, overwriting anything so far - so someone can overwrite the Authentication: header if they want by passing it via dispatchOptions...

File: lib/dispatcher.js [hunk 3]
```diff
@@ -151,7 +151,7 @@ Dispatcher.prototype.dispatch = function(params, dispatchOptions) {
...
       me._addVersionInfo(params);
```
Comment (praecipula) [inline lib/dispatcher.js]: Then we add the version header here, meaning that users of the library *can't* overwrite this header - which I'm totally cool with. Just an interesting thing I noticed here, but I think we can keep this behavior?

Comment (praecipula) [reply lib/dispatcher.js]: Also, aside: any idea why Github chooses to place comments on the left vs. right?

File: lib/dispatcher.js (hunk 2 above)
Comment (joetrollo) [reply lib/dispatcher.js]: That's true, but rare, and they should know what they're doing if that's what they want.

File: lib/dispatcher.js (hunk 3 above)
Comment (joetrollo) [reply lib/dispatcher.js]: Yes, I think we should keep it. Also, GitHub puts it on the left when the lines aren't changed.

**Pull request merged**